ui/app/AppLayouts/Wallet/popups/filterSubMenus/ActivityTypeFilterSubMenu.qml

```diff
@@ -3,31 +3,21 @@ import QtQuick.Controls 2.15
 
 import StatusQ.Popups 0.1
 
+import utils 1.0
+
 import "../../controls"
 
 StatusMenu {
     id: root
 
-    property var typeFilters:[]
+    property var typeFilters: []
 
     signal back()
-    signal actionTriggered(int action, bool checked)
-
-    property bool allChecked: {
-        let allCheckedIs = true
-        for(var i=0;i< root.contentChildren.length;i++) {
-            if(root.contentChildren[i].checkState === Qt.Unchecked)
-                allCheckedIs = false
-        }
-        return allCheckedIs
-    }
+    signal actionTriggered(int type)
 
-    enum TxType {
-        Send,
-        Receive,
-        Buy,
-        Swap,
-        Bridge
+    QtObject {
+        id: d
+        readonly property bool allChecked: typeFilters.length === 5
```

Comment:
This magic hardcoded number will once bite back...

Reply:
I found a way, not the best but I iterate through the children and look for all the checkboxes and it works okay @Cuteivist @caybro , please approve when you can :D